src/main/java/mate/academy/service/ShoppingCartServiceImpl.java

```diff
@@ -16,53 +16,55 @@
     private final BookRepository bookRepository;
     private final CartItemRepository cartItemRepository;
-    private final CartItemMapper cartItemMapper;
     private final UserRepository userRepository;
+    private final CartItemMapper cartItemMapper;
+    private final ShoppingCartMapper shoppingCartMapper;
 
     @Override
-    public void addItemToCart(Authentication authentication,
-                              CartItemRequestDto cartItemRequestDto) {
-        User user = getUser(authentication);
+    @Transactional
+    public void addItemToCart(CartItemRequestDto cartItemRequestDto) {
+        User user = (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
         ShoppingCart shoppingCart = shoppingCartRepository.findShoppingCartByUserId(user.getId())
                 .orElseGet(() -> registerNewShoppingCart(user));
 
         Optional<CartItem> existingCartItem = shoppingCart.getCartItems().stream()
                 .filter(item -> item.getBook().getId().equals(cartItemRequestDto.getBookId()))
                 .findFirst();
-        CartItem cartItemToUpdate;
-        if (existingCartItem.isPresent()) {
-            cartItemToUpdate = existingCartItem.get();
-            cartItemToUpdate.setQuantity(cartItemRequestDto.getQuantity()
-                    + cartItemToUpdate.getQuantity());
-        } else {
-            cartItemToUpdate = createNewCartItem(cartItemRequestDto, shoppingCart);
-        }
-        cartItemRepository.save(cartItemToUpdate);
+        final CartItem[] cartItemToUpdate = new CartItem[1];
+        existingCartItem.ifPresentOrElse(
+                cartItem -> {
+                    cartItem.setQuantity(cartItemRequestDto.getQuantity() + cartItem.getQuantity());
+                    cartItemToUpdate[0] = cartItem;
+                },
+                () -> cartItemToUpdate[0] = createNewCartItem(cartItemRequestDto, shoppingCart)
+        );
+        cartItemRepository.save(cartItemToUpdate[0]);
     }
 
     @Override
-    public ShoppingCartDto getAllCartItems(Authentication authentication) {
-        User user = getUser(authentication);
+    @Transactional
+    public ShoppingCartDto getAllCartItems() {
+        User user = (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
         ShoppingCart shoppingCart = shoppingCartRepository.findShoppingCartByUserId(user.getId())
                 .orElseGet(() -> registerNewShoppingCart(user));
         return shoppingCartMapper.toDto(shoppingCart);
     }
 
     @Override
-    public void updateBookQuantity(Authentication authentication,
-                                   Long cartItemId,
+    @Transactional
+    public void updateBookQuantity(Long cartItemId,
                                    CartItemQuantityRequestDto qtyRequestDto) {
-        User user = getUser(authentication);
+        User user = (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
```

Comment:
No, your bussiness logic shouldn't depend on infrastructure. You need to User instead of Authefication to your Services